packages/sdk/src/tasks/deposit-erc20.ts

```diff
@@ -405,13 +414,15 @@ task('deposit-erc20', 'Deposits WETH9 onto L2.')
       }
     }
 
-    const postBalance = await WETH9.balanceOf(signer.address)
-
-    const expectedBalance = preBalance.add(utils.parseEther('1'))
-    if (!expectedBalance.eq(postBalance)) {
-      throw new Error(
-        `Balance mismatch, expected: ${expectedBalance}, actual: ${postBalance}`
-      )
+    if (args.checkBalanceMismatch) {
+      const postBalance = await WETH9.balanceOf(address)
+      const expectedBalance = preBalance.add(parseEther('1'))
```

Comment:
```diff
-const expectedBalance = preBalance.add(parseEther('1'))
+const expectedBalance = preBalance!.add(parseEther('1'))
```

Reply:
I think the non-null assertion `!` is not necessary, cause the compiler does not show any error without it.